pages/api/rpc.ts

```diff
@@ -1,53 +1,86 @@
-import type { NextApiRequest, NextApiResponse } from 'next';
-import { getRPCUrls } from '@lido-sdk/fetch';
-import { CHAINS } from '@lido-sdk/constants';
-import {
-  wrapRequest,
-  defaultErrorHandler,
-} from '@lidofinance/next-api-wrapper';
 import getConfig from 'next/config';
-import { fetchWithFallbacks } from 'utils/fetchWithFallbacks';
-import { serverLogger } from 'utils/serverLogger';
+import { externalRPC } from 'config';
+import { NextApiRequest, NextApiResponse } from 'next';
+import { fetchRpc, FetchRpcInitBody, iterateUrls } from '@lidofinance/rpc';
+import { serverLogger } from 'utilsApi';
 
-const { serverRuntimeConfig } = getConfig();
-const { infuraApiKey, alchemyApiKey, apiProviderUrls } =
-  serverRuntimeConfig as RuntimeConfig;
+// This code is copy of https://github.com/lidofinance/warehouse/tree/main/packages/next/pages
```

Comment:
Why do you copy code and don't use rpcFactory? 🤔

Reply:
@DiRaiks there is a comment on the next line

> This code is copy of ... but metrics were removed

The current implementation of `rpcFactory` only works with metrics, but there are no metrics in this project right now. So there were two options
1) update package to work without metrics
2) copy package and remove metrics usage
3) somehow mock but not actually use metrics

I've picked the second option, because we probably will add metrics at some point, so this code can be replaced with `rpcFactory`. Also, if we won't actually add metrics, we can go and update the package. So it feels like this approach is a bit more versatile.